crates/bevy_pbr/src/render/light.rs

```diff
@@ -378,6 +379,25 @@ impl SpecializedMeshPipeline for ShadowPipeline {
     }
 }
 
+#[derive(Component)]
+pub struct ExtractedAmbientLight {
+    ambient_color: Vec4,
+}
+
+pub fn extract_ambient_light(
+    mut commands: Commands,
+    views: Extract<Query<(Entity, &AmbientLight), With<Camera>>>,
+) {
+    for (entity, ambient_light) in &views {
+        commands
+            .get_or_spawn(entity)
+            .insert((ExtractedAmbientLight {
+                ambient_color: Vec4::from_slice(&ambient_light.color.as_linear_rgba_f32())
+                    * ambient_light.brightness,
+            },));
+    }
```

Comment:
This moves the computation of the light to the `Extract` stage. This will slow `Extract` which should be made as fast as possible, because it will be the only non-parallel bit of the pipelined rendering update.  I'd recommend you keep it where it was before. No need to define a `ExtractedAmbientLight`, just push the component as-is.